src/tests/functional/shared_test_classes/src/single_op/non_max_suppression.cpp

```diff
@@ -148,5 +148,251 @@ void Nms9LayerTest::SetUp() {
     function = std::make_shared<ov::Model>(nms, params, "NMS");
 }
 
+
+void NmsLayerTest::compare(const std::vector<ov::Tensor>& expected, const std::vector<ov::Tensor>& actual) {
+    CompareBBoxes(expected, actual);
+}
+
+
+namespace {
+
+typedef struct Rect {
+    int32_t x1;
+    int32_t y1;
+    int32_t x2;
+    int32_t y2;
+} Rect;
+
+class Box {
+public:
+    Box() = default;
+
+    Box(int32_t batchId, int32_t classId, int32_t boxId, Rect rect, float score) {
+        this->batchId = batchId;
+        this->classId = classId;
+        this->boxId = boxId;
+        this->rect = rect;
+        this->score = score;
+    }
+
+    int32_t batchId;
+    int32_t classId;
+    int32_t boxId;
+    Rect rect;
+    float score;
+};
+
+template <typename fromT, typename toT>
+void convert(fromT* src, toT* dst, size_t size) {
```

Comment:
Can we used `ov::reference::convert` instead?
It should support all OV types.

Reply:
ov::reference::convert has no advantages here. I would use it in case of small types (<1byte)

Reply:
It was suggestion to re-use code not write it again.

Reply:
What about create some kind of unified util func to convert ov::Tensors or just std::vector from one type to another.
It may really helps to clean up code.